### PR TITLE
Add validations for ServiceClass and ServicePlan

### PR DESCRIPTION
**Pull request by pmorie**
Part of #268; adds validations for `ServiceClass` and `ServicePlan`.

### CODE REVIEW

File: pkg/apis/servicecatalog/validation/broker.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func validateBrokerSpec(spec *sc.BrokerSpec, fldPath *field.Path) field.ErrorLis
 
 	if "" == spec.URL {
 		allErrs = append(allErrs,
-			field.Required(fldPath.Child("URL"),
+			field.Required(fldPath.Child("url"),
```
Comment (arschles) [inline pkg/apis/servicecatalog/validation/broker.go]: can you put this and the above into `const`s?

Comment (pmorie) [inline pkg/apis/servicecatalog/validation/broker.go]: Normally I would be totes onboard with that, but i'm not sure that makes things more readable in this case.

Comment (pmorie) [inline pkg/apis/servicecatalog/validation/broker.go]: I'm happy to come back to it once the dust settles

**Comment by arschles** [comment]
I'm merging this now. I don't see any significant reservations with it, and having validations in the controller is important for our MVP releases.

**Pull request merged**